### ISSUE TITLE
failure to verify that all objects have `id` field when running `cargo test` in sui-framework

### ISSUE THREAD

**Issue by emmazzz** [MystenLabs/sui#4268]
## Steps to Reproduce Issue

First, add the following Move code to `crates/sui-framework/sources`:
```move
module sui::m {
    use sui::transfer;
    #[test_only]
    use sui::test_scenario;

    struct S has key {
        // no id field here
        x: u64
    }

    fun create_and_transfer() {
        transfer::transfer(S { x: 1 }, @0x42); 
    }

    #[test]
    fun test() {
        let scenario = &mut test_scenario::begin(&@0x42);
        create_and_transfer();
        test_scenario::next_tx(scenario, &@0x42);
        let s = test_scenario::take_owned<S>(scenario);
        test_scenario::return_owned(scenario, s);
    }
}
```
Then, run `cargo test` in `crates/sui-framework`. This will run the Move unit tests including the one we just added.
## Expected Result

Some nice error message telling me that I should add an `id` field to `S`.
## Actual Result
```
failures:

---- tests::cmework_move_unit_tests stdout ----
thread '<unnamed>' panicked at 'called `Result::unwrap()` on an `Err` value: PartialVMError { major_status: 
INTERNAL_TYPE_ERROR, sub_status: None, message: Some("cannot cast U64(1) to struct"), exec_state: None, indices: [], 
offsets: [] }', crates/sui-framework/src/natives/test_scenario.rs:72:59
```

**Comment by emmazzz** [comment]
IIUC the issue is that `cargo test` calls `build_move_package`. Instead it should be calling `build_and_verify_package` 
https://github.com/MystenLabs/sui/blob/main/crates/sui-framework/src/lib.rs#L223-L226

**Comment by emmazzz** [comment]
Resolved by #4342
